pkg/kapp/cmd/core/config_factory.go

```diff
@@ -113,7 +113,12 @@ func (f *ConfigFactoryImpl) clientConfig() (bool, clientcmd.ClientConfig, error)
 	}
 
 	if len(configYAML) > 0 {
-		envHostPort := net.JoinHostPort(os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT"))
+		// If it is not provided, we default the Kubernetes service port to 443
+		kubernetesServicePort, ok := os.LookupEnv("KUBERNETES_SERVICE_PORT")
```

Comment:
`ok` will be true if the environment variable is set to empty string (which can be the case sometimes), so I think we can just go ahead with `Getenv` and checking on empty string.

Reply:
I didn't think of that!